Src/Shaders/PBREffect.fx

```diff
@@ -0,0 +1,284 @@
+VSOutputPixelLightingTxTangent VSConstantBn(VSInputNmTxTangent vin)
+{
+    VSOutputPixelLightingTxTangent vout;
+
+    float3 normal = BiasX2(vin.Normal);
+
+    CommonVSOutputPixelLighting cout = ComputeCommonVSOutputPixelLighting(vin.Position, normal);
+
+    vout.PositionPS = cout.Pos_ps;
+    vout.PositionWS = float4(cout.Pos_ws, 1);
+    vout.NormalWS = cout.Normal_ws;
+    vout.Diffuse = float4(ConstantAlbedo, Alpha);
+    vout.TexCoord = vin.TexCoord;
+
+    float3 tangent = BiasX2(vin.Tangent.xyz);
+    vout.TangentWS = normalize(mul(tangent, WorldInverseTranspose));
+
+    return vout;
+}
+
+
+// Vertex shader: pbr + velocity (biased normal)
+VSOut_Velocity VSConstantVelocityBn(VSInputNmTxTangent vin)
+{
+    VSOut_Velocity vout;
+
+    float3 normal = BiasX2(vin.Normal);
+
+    CommonVSOutputPixelLighting cout = ComputeCommonVSOutputPixelLighting(vin.Position, normal);
+
+    vout.current.PositionPS = cout.Pos_ps;
+    vout.current.PositionWS = float4(cout.Pos_ws, 1);
+    vout.current.NormalWS = cout.Normal_ws;
+    vout.current.Diffuse = float4(ConstantAlbedo, Alpha);
+    vout.current.TexCoord = vin.TexCoord;
+
+    float3 tangent = BiasX2(vin.Tangent.xyz);
+    vout.current.TangentWS = normalize(mul(tangent, WorldInverseTranspose));
+
+    vout.prevPosition = mul(vin.Position, PrevWorldViewProj);
+
+    return vout;
+}
+
+
+// Pixel shader: pbr (constants) + image-based lighting
+float4 PSConstant(PSInputPixelLightingTxTangent pin) : SV_Target0
+{
+    // vectors
+    const float3 V = normalize(EyePosition - pin.PositionWS.xyz);   // view vector
+    const float3 N = normalize(pin.NormalWS);                       // surface normal
+    const float AO = 1;                                             // ambient term
+
+    float3 color = LightSurface(V, N, 3,
+        LightColor, LightDirection,
+        ConstantAlbedo, ConstantRoughness, ConstantMetallic, AO);
+
+    return float4(color, Alpha);
+}
+
+
+// Pixel shader: pbr (textures) + image-based lighting
+float4 PSTextured(PSInputPixelLightingTxTangent pin) : SV_Target0
+{
+    const float3 V = normalize(EyePosition - pin.PositionWS.xyz); // view vector
+    const float3 L = normalize(-LightDirection[0]);               // light vector ("to light" opposite of light's direction)
+
+    // Before lighting, peturb the surface's normal by the one given in normal map.
+    float3 localNormal = BiasX2(NormalTexture.Sample(SurfaceSampler, pin.TexCoord).xyz);
+    float3 N = PeturbNormal(localNormal, pin.NormalWS, pin.TangentWS);
+
+    // Get albedo
+    float4 albedo = AlbedoTexture.Sample(SurfaceSampler, pin.TexCoord);
+
+    // Get roughness, metalness, and ambient occlusion
+    float3 RMA = RMATexture.Sample(SurfaceSampler, pin.TexCoord);
+
+    // glTF2 defines metalness as B channel, roughness as G channel, and occlusion as R channel
+
+    // Shade surface
+    float3 color = LightSurface(V, N, 3, LightColor, LightDirection, albedo.rgb, RMA.g, RMA.b, RMA.r);
```

Comment:
The original SimplePBR12 sample used the Roughness (R), Metalness (G), Ambient-occlusion (B) channel order. glTF2 defines it as Ambient-Occlusion (R), Roughness (G), Metalness (B), so I'm using that for DirectX Tool Kit as a "standard".